codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlShapeDeserVisitor.java

```diff
@@ -69,21 +71,21 @@ public XmlShapeDeserVisitor(GenerationContext context, Predicate<MemberShape> me
         return Collections.singletonMap("decoder", "smithyxml.NodeDecoder");
     }
 
-    private XmlMemberDeserVisitor getMemberDeserVisitor(MemberShape member, String dataDest, boolean isXmlAttibuteMember) {
+    private XmlMemberDeserVisitor getMemberDeserVisitor(MemberShape member, String dataDest, boolean isXmlAttributeMember) {
         // Get the timestamp format to be used, defaulting to rfc 3339 date-time format.
         TimestampFormatTrait.Format format = member.getMemberTrait(getContext().getModel(), TimestampFormatTrait.class)
                 .map(TimestampFormatTrait::getFormat).orElse(DEFAULT_TIMESTAMP_FORMAT);
-        // Check if FlattenedTrait  is applied
-        boolean isFlattened = member.hasTrait(XmlFlattenedTrait.ID);
-        return new XmlMemberDeserVisitor(getContext(), dataDest, format, isXmlAttibuteMember, isFlattened);
+        return new XmlMemberDeserVisitor(getContext(), member, dataDest, format, isXmlAttributeMember);
     }
 
     // generates code to define and initialize output variable for an aggregate shape
-    private void generatesIntializerForOutputVariable(GenerationContext context, Shape shape) {
+    private void generatesInitializerForOutputVariable(GenerationContext context, Shape shape) {
         GoWriter writer = context.getWriter();
         Symbol shapeSymbol = context.getSymbolProvider().toSymbol(shape);
 
         writer.write("var sv $P", shapeSymbol);
+        // TODO [denseListMap] need to reference if pointable type. Assumes containers are always double pointers.
+        // TODO this probably isn't true for union types.
```

Comment:
still todo's?

Reply:
I thought I removed all of those, must of missed one, thanks!

Reply:
This comment was wrong, union de-serialization will always be a struct that satisfy the union member interface. And all types in this context will be pointers like list/maps, or double pointers like structs.